packages/ember-testing/lib/ext/application.js

```diff
@@ -76,7 +76,7 @@ EmberApplication.reopen({
 
     this.testing = true;
 
-    this.Router.reopen({
+    this.resolveRegistration('router:main').reopen({
```

Comment:
No, I don't think we can. I played with a few different ways to support re-registration and none of them were going very well.

This change, practically, means that you *must* call `App.setupForTesting()` *after* you set a custom router (usually via `App.Router = Router.extend(`). I updated the docs for `setupForTesting` to make this clear, it was already implied by the `reopen` call and the docs, but was hopelessly coupled to the globals resolver.

Also in practice, this `reopen` doesn't matter. In Ember-CLI apps (most apps) we set `location: 'none'` via the `env` anyway. So there is some dubious stuff going on here.

Perhaps instead of forcing `location: 'none'`, we should warn or assert if the location is something other than `none`.

I'm going to call this good for canary.